sweet/harnesses/cockroachdb.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package harnesses
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"golang.org/x/benchmarks/sweet/common"
+	"golang.org/x/benchmarks/sweet/common/log"
+)
+
+// CockroachDB implements the Harness interface.
+type CockroachDB struct{}
+
+func (h CockroachDB) CheckPrerequisites() error {
+	// Cockroachdb is only supported on arm64 and amd64 architectures.
+	if runtime.GOARCH != "arm64" && runtime.GOARCH != "amd64" {
+		return fmt.Errorf("requires amd64 or arm64")
+	}
+	return nil
+}
+
+func (h CockroachDB) Get(gcfg *common.GetConfig) error {
+	// Build against the latest stable release.
+	// Deep clone the repo as we need certain submodules, i.e.
+	// PROJ, for the build to work.
+	return gitDeepClone(
+		gcfg.SrcDir,
+		"https://github.com/cockroachdb/cockroach",
+		"v24.1.0-rc.1",
```

Comment:
What is the version upgrade story with `sweet`? Will we periodically update this version and check whether any benchmark results change?